x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -181,7 +181,8 @@
         this.description = description;
         this.tags = Collections.unmodifiableList(ExceptionsHelper.requireNonNull(tags, TAGS));
         this.metadata = metadata == null ? null : Collections.unmodifiableMap(metadata);
-        this.input = ExceptionsHelper.requireNonNull(input, INPUT);
+        this.input = (input == null && modelType == TrainedModelType.PYTORCH) ?
```

Comment:
If this is going to be our design, we should have the input default based on the model type, and consequently set there. 

```
this.input = ExceptionsHelper.requireNonNull(input == null ? modelType.defaultInput() : input, INPUT);
```